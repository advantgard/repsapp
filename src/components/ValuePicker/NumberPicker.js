import React from "react";
import { ValuePicker } from "./ValuePicker";
import { createNumberDigits } from "../../lib/ValuePickerHelpers";

export const NumberPicker = props => {

	const value = createNumberDigits( props.value, props.maxValue );
	const maxOperationValue = Math.pow( 10, value.length - 1 );

	const digits = value.split( "" ).map( ( value, index ) => {

		const divisor = index ? Math.pow( 10, index ) : 1;
		const operationValue = maxOperationValue / divisor;

		return <ValuePicker
			increase={props.handler.bind( null, "increase", operationValue )}
			decrease={props.handler.bind( null, "decrease", operationValue )}
			value={value}
			key={index}
		/>;

	} );

	return <div className="number-picker"> {digits} </div>;

};