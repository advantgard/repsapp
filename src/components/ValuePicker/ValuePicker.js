import React from "react";

export const ValuePicker = props =>

	<div className="value-picker">
		<button className="value-picker__button--increase" onClick={props.increase}/>
		<div className="value-picker__box"> {props.value} </div>
		<button className="value-picker__button--decrease" onClick={props.decrease}/>
	</div>;