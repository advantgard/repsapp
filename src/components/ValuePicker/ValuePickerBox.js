import React from "react";

export const ValuePickerBox = props =>

	<div className="value-picker__box">{props.value}</div>;