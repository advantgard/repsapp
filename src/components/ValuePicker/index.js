export { ValuePicker } from "./ValuePicker";
export { NumberPicker } from "./NumberPicker";