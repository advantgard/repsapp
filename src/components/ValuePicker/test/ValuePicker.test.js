import React from "react";
import renderer from "react-test-renderer";
import { ValuePicker } from "../ValuePicker";

describe( "ValuePicker", () => {

	test( "renders correctly", () => {

		const tree = renderer.create(
			<ValuePicker value="0" increase={() => {
			}} decrease={() => {
			}}/>
		).toJSON();

		expect( tree ).toMatchSnapshot();

	} );

} );