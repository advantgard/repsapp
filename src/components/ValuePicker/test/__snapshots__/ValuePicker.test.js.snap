// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ValuePicker renders correctly 1`] = `
<div
  className="value-picker"
>
  <button
    className="value-picker__button--increase"
    onClick={[Function]}
  />
  <div
    className="value-picker__box"
  >
     
    0
     
  </div>
  <button
    className="value-picker__button--decrease"
    onClick={[Function]}
  />
</div>
`;
